### PR TITLE
daos: improvements

### LINKED ISSUE #448: daos: add dsync nd dcmp documentation to DAOS-Support.md
DAOS dsync and dcmp documentation should be added to [DAOS-Support.md](https://github.com/hpc/mpifileutils/blob/master/DAOS-Support.md)

### PR DESCRIPTION
**Pull request by daltonbohning**
Two commits:
1. Improve some of the path parsing by NOT using static buffers, NOT limiting paths to PATH_MAX explicitly, and using the "n" versions of some string functions for buffer overflow protection.
2. Added documentation for dsync and dcmp

closes #448

### CODE REVIEW

File: DAOS-Support.md [hunk 1]
```diff
@@ -98,7 +100,7 @@ $ mpirun -np 3 dcp -v /tmp/$USER/conts/p1cont1 /tmp/$USER/d
 [2020-04-23T17:17:51]   Files: 3
 [2020-04-23T17:17:51]   Links: 0
 ```
-## DAOS non-POSIX Data Movement Examples 
+## DAOS non-POSIX Data Movement Examples With dsync
```
Comment (dsikich) [inline DAOS-Support.md:104]: @daltonbohning just a minor typo: "With" vs. "with"

Comment (daltonbohning) [reply DAOS-Support.md:104]: @dsikich I don't mind either way. _**Technically**_, I think different style guides (MLA, APA, Chicago, etc.) have different rules about capitalizing "with" in a title. I'll go with the lowercase version as you suggest. (And to be consistent with other usage)

Comment (dsikich) [reply DAOS-Support.md:104]: @daltonbohning yeah, not a big deal I guess I'm just used to it being lowercase.

**Review by dsikich — APPROVED** [review]
@daltonbohning looks good.

File: DAOS-Support.md [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+filesystems in dcp and dsync:
 
 1. DAOS  -> POSIX
 2. POSIX -> DAOS
 3. DAOS  -> DAOS
 
-The DAOS->DAOS case supports both POSIX and non-POSIX DAOS containers.
+For dcp, the DAOS->DAOS case supports both POSIX and non-POSIX DAOS containers.
+
+For dsync, the DAOS->DAOS case currently only supports POSIX containers.
```
Comment (adammoody) [inline DAOS-Support.md:15]: Where does dcmp fall?

**Review by adammoody — APPROVED** [review]
Thanks @daltonbohning

**Comment by daltonbohning** [comment]
Thanks!

**Pull request merged**